lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -944,10 +958,26 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
              cast<ValueDecl>(D)->isInstanceMember()))
           continue;
         
-        if (isa<FuncDecl>(candidate) &&
-            (cast<FuncDecl>(candidate)->getParameters()->size() !=
-             cast<FuncDecl>(D)->getParameters()->size()))
-          continue;
+        if (isa<FuncDecl>(candidate)) {
+          auto cParams = cast<FuncDecl>(candidate)->getParameters();
+          auto dParams = cast<FuncDecl>(D)->getParameters();
+          
+          if (cParams->size() != dParams->size())
+            continue;
+          
+          bool hasSameParameterTypes = true;
+          for (auto index : indices(*cParams)) {
+            if (cParams->get(index)->getType()->getCanonicalType() !=
+                dParams->get(index)->getType()->getCanonicalType()) {
```

Comment:
Hm…I think checking canonical types is probably too strict. What if someone changes optionality, or uses a superclass instead of a subclass? `TypeBase::matchesParameter` might be a good thing to look at here, probably using the `AllowOverride` and `AllowTopLevelOptionalMismatch` flags.

Another option would be to only worry about matching types if there are multiple overloads. That both saves work and offers a probably-better experience in the weird cases I mentioned above.

Reply:
Thank you for the suggestion, I'll update this soon